examples/app_blink.rs

```diff
@@ -0,0 +1,33 @@
+#![feature(plugin, no_std, core, start)]
+#![no_std]
+#![plugin(macro_zinc)]
+
+extern crate core;
+extern crate zinc;
+
+use zinc::hal::timer::Timer;
+use zinc::hal::lpc17xx::{pin, timer};
+use zinc::hal::pin::GpioDirection;
+use zinc::hal::pin::Gpio;
+use core::option::Option::Some;
+
+#[zinc_main]
```

Comment:
Are you able to rollback the `#[zinc_main]` changes. @farcaller has a bit on at the moment and adding a new syntax extension is just going to make it harder for him to get the review completed and this merged.

I like the decorator macro though, well done.